decepticonlp/metrics/temporal_metrics.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torch.nn as nn
+
+
+class RankCharacters:
+    """
+        Accepts a feature vector torch tensor and outputs a temporal ranking of characters
+        inputs of shape (number of words,feature_vector_size)
+    """
+
+    def temporal_score(self, sentence):
+        """ 
+            Considering a input sequence x1,x2,...,xn
+            we calculate T(xi) = F(x1,x2,...,xi) - F(x1,x2,...,xi-1)
+            where F is one pass through a RNN cell 
+        """
+        inputs = sentence.reshape(len(sentence), 1, -1)
+        model = nn.RNN(inputs.shape[2], 1)
+        with torch.no_grad():
+            _, pred = model(inputs)
```

Comment:
Shouldn't the model be pre-trained on the task? I mean user should have an option to use his own models to test this

Reply:
@rajaswa , this is the temporal scorer, it selects words based on highest decrease in rnn outputs.

Reply:
1. I am aware of that. But monitoring output from an untrained RNN doesn't make sense at all
2. This should be usable by all models working with text sequences

Reply:
Cool I'll make the changes in that way then !